lib/seahorn/BvOpSem2Allocators.cc

```diff
@@ -296,7 +301,15 @@ class StaticOpSemAllocator : public OpSemAllocator {
       preAlloc(inst, memSz, true);
     } else {
       // -- allocate max allowed for dynamically sized allocations
-      preAlloc(inst, m_maxSymbAllocSz, false);
+      // TODO: use rng info here
+      auto operand_inst = dyn_cast<llvm::Instruction>(inst.getOperand(0));
+      auto rangeVal = m_sem.getInstRng(*operand_inst);
+      unsigned memSz =
+          rangeVal.second != 0 ? rangeVal.second : m_maxSymbAllocSz;
+      LOG("bmc-crab", MSG << "Range for " << *operand_inst << " = ["
+                          << rangeVal.first << "," << rangeVal.second
+                          << "]\n";);
+      preAlloc(inst, memSz, false);
```

Comment:
I do not see how allocation based on unknown upper bound is handled. Whether the assumption that not more is allocated at run time is added or not

Reply:
this comment remains unaddressed and is still relevant. If we are using a bound on allocation size, we must mark it and add appropriate assumption in vcgen. I believe the 3rd arugment to `preAlloc` controls that.

Reply:
I'm confused here. Crab could infer an upper bound for allocation, but that bound is not fixed size here. For example, the test I was given contains a range of allocation [4,400] for variable `size`, and the size for allocation is not fixed except variable `len` is determined.